cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -362,7 +362,7 @@ func (csr *ClusterStateRegistry) IsClusterHealthy() bool {
 	csr.Lock()
 	defer csr.Unlock()
 
-	totalUnready := csr.totalReadiness.Unready + csr.totalReadiness.LongNotStarted + csr.totalReadiness.LongUnregistered
```

Comment:
What is the reasoning for removing LongNotStarted and LongUnregistered from the condition?

Reply:
LongNotStarted and LongUnregistered are nodes that failed to start. In my opinion such nodes should be removed and other mechanism will backoff node pool, Cluster Autoscaler should recover from situation when there is a lot of such nodes. Deleting such nodes is not a problem as we are sure that nothing is running on these nodes. Cluster should be considered as unhealthy only when large part of nodes that were ready becomes unready because there are some pods running on these nodes and we don't want to break it even more.

Reply:
Ack, didn't realize node group health remains unchanged.